crates/sui-types/src/message_envelope.rs

```diff
@@ -101,6 +101,9 @@ pub trait Message {
 
     fn digest(&self) -> Self::DigestType;
 
+    /// Perform cheap validity checks before any expensive crypto verification.
+    fn verify_user_input(&self) -> SuiResult;
```

Comment:
This method is only applicable to transactions. What's the reason to add it to envelope instead of just transaction?